### ISSUE TITLE
Allow for file uploads

### ISSUE THREAD

**Issue by coreation** [tdt/core#318]
File uploads allow people with datasets to upload their files to the server in order to publish them. This takes away the step of that person needing to publishing the file somewhere online before publishing it.

**Comment by pietercolpaert** [comment]
I think over here we should certainly enable the option for cloud storage. Your thoughts?

How about using http://flysystem.thephpleague.com/ for this?

**Comment by coreation** [comment]
I was going to do that yes, a link towards #311 is welcome I think.

**Comment by wesleydv** [comment]
Why is this remove from TDT6.0 it is an urgent requirement for Digipolis.

**Comment by coreation** [comment]
Is it? I thought Digipolis had a special own place for all of their files to which they could FTP or something?

**Comment by daften** [comment]
Yes, could be, but we need file upload in the interface. The goal is to remove data management from the portal and put this in TDT, but for that we need file upload in the UI.
The API would be used as a read-only API then.

**Comment by coreation** [comment]
Are you guys planning on uploading the 200MB files as well? Because that would mean allowing to post such a large file + allowing the connecting to be open that long?

**Comment by daften** [comment]
Yes, but that is server settings and can be configured. First thing is the functionality :)

**Comment by coreation** [comment]
Planned for the next release it is :) Will probably be implemented with an internal deadline somewhere in september I believe.

**Comment by pietercolpaert** [comment]
I'd consider:
 * Using flysystem for storing the data on different kind of locations: http://flysystem.thephpleague.com/
 * Implementing an option to circumvent The DataTank to do magic with the file when the representation is big and cannot be loaded into memory.

**Comment by coreation** [comment]
* Flysystem is on our radar, we use it in other projects to connect with Dropbox and Google Drive (@daften is that an option?)
* You mean only showing a chunk of the data, like we're doing now or?

**Comment by pietercolpaert** [comment]
* See #311 
 * See #258

**Comment by daften** [comment]
Ik ken flysystem niet, maar dat ziet er wel OK uit, dus sure :)

**Comment by coreation** [comment]
Flysystem is a league package, big fan :) So we can provide a configuration somewhere to add an application key of GDrive or Dropbox in order to define paths to data files?

**Comment by daften** [comment]
I think local file storage should also be pluggable (IMO).
Remote filesystems can be cloud tools or nfs shares maybe?

**Comment by coreation** [comment]
Flysystem abstract filesystems entirely, so it doesn't really matter. If we use the local file storage however, we need to build an upload functionality though. So either we focus on local, or Dropbox, Gdrive, ... . The implementation will be using Flysystem but if we know which implementation we can focus on, that'd be nice ;)

**Comment by daften** [comment]
Vraag ik om door onze analyste te laten bepalen dan ;)



--
Dieter Blomme
